### PR TITLE
User signup

### PR DESCRIPTION
**Pull request by lnevado**
## Summary

- Added user signup with password (alphanumeric 8+ characters) and email (wolox domain and unique) requirements.
- Added logging of custom errors and user creation.

## Trello Card
https://trello.com/c/quqSdje0/2-sign-up

### CODE REVIEW

**Review by igna92ts — COMMENTED** [review]
Build is failing

File: app/middlewares/errors.js [hunk 1]
```diff
@@ -4,6 +4,8 @@ const errors = require('../errors'),
 const DEFAULT_STATUS_CODE = 500;
 
 const statusCodes = {
+  [errors.INVALID_EMAIL]: 400,
+  [errors.ENCRYPTION_ERROR]: 500,
   [errors.INVALID_USER]: 400,
   [errors.BOOK_NOT_FOUND]: 404,
```
Comment (igna92ts) [inline app/middlewares/errors.js]: remove example error

File: test/app.js [hunk 2]
```diff
@@ -1,44 +0,0 @@
-'use strict';
```
Comment (igna92ts) [inline test/app.js]: why did you delete this?

**Review by igna92ts — CHANGES_REQUESTED** [review]

File: app/middlewares/errors.js (hunk 1 above)
Comment (lnevado) [reply app/middlewares/errors.js]: Removed example errors

File: test/app.js (hunk 2 above)
Comment (lnevado) [reply test/app.js]: Re-added it

**Review by igna92ts — APPROVED** [review]

**Pull request merged**